libs/wingsdk/src/target-sim/redis.inflight.ts

```diff
@@ -0,0 +1,79 @@
+import Dockerode from "dockerode";
+import { ISimulatorResourceInstance } from "./resource";
+import { RedisSchema } from "./schema-resources";
+import { IRedisClient } from "../redis";
+import { ISimulatorContext } from "../testing/simulator";
+
+// Issue using types from ioredis with JSII
+//import { Redis as IoRedis } from "ioredis";
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const IoRedis = require("ioredis");
+
+export class Redis implements IRedisClient, ISimulatorResourceInstance {
+  static uid = 0;
+  private readonly container_name: string;
+  private readonly context: ISimulatorContext;
+
+  private connection_url?: string = undefined;
+  private docker?: Dockerode = undefined;
+  private connection?: any;
+
+  public constructor(_props: RedisSchema["props"], context: ISimulatorContext) {
+    this.context = context;
+    this.container_name = `wing-sim-redis-${this.context.resourcePath.replace(
+      "/",
+      "."
+    )}-${Redis.uid}`;
+    Redis.uid++;
+  }
+
+  public async init(): Promise<void> {
+    this.docker = new Dockerode();
+
+    // Create a redis container
+    try {
+      let container = await this.docker.createContainer({
+        Image: "redis",
+        name: this.container_name,
+      });
```

Comment:
```suggestion
      let container = await this.docker.createContainer({
        Image: "redis",
        name: this.container_name,
        ExposedPorts: { "6379/tcp": {} },
      });
```

This is my guess of what could help make sure the container is accessible.

Reply:
hmm it looks like if you are using a github action environment for your actions then any started containers need to be accessed using `localhost:<port>` or `127.0.0.1:<port>` but when your actions are run within a container they have access to the user defined docker bridge network.

Reply:
I don't think explicitly exposing the ports will help, 6379/tcp is already exposed in the dockerfile of the redis image so I doubt it'll make a change. Nonetheless I'll give it a try.

> when your actions are run within a container they have access to the user defined docker bridge network.

Any chance we can change our actions to be run in a container to check this theory?

Reply:
> I don't think explicitly exposing the ports will help, 6379/tcp is already exposed in the dockerfile of the redis image so I doubt it'll make a change. Nonetheless I'll give it a try.

Tried, same issue, reverted.

Reply:
> Any chance we can change our actions to be run in a container to check this theory?

It doesn't look too bad to try. Im not GHA expert, it looks simple enough. Ill test in separate junk PR.

Reply:
:pray: thanks.

Reply:
shame on me for thinking itd be so simple, keep running into tiny hitches with permissions and such but making progress, shooting to get it working this evening.

Reply:
@yoav-steinberg @MarkMcCulloh I finally built a docker container with everything needed to get to the build and tests going. Im not sure if the results are promising or not, I am failing for a separate reason.

https://github.com/winglang/wing/actions/runs/4452587048/jobs/7820349387?pr=1842

<img width="1011" alt="image" src="https://user-images.githubusercontent.com/45375125/226075437-d9e97cfc-4ac8-4f96-8ff9-dfb90ac7d857.png">

Gonna look into it more this weekend

Reply:
So after an idea to use port mappings and local hosts I got this working (not in contianer)
https://github.com/winglang/wing/actions/runs/4491143694/jobs/7899192318